pkg/cache/v3/snapshot.go

```diff
@@ -62,6 +86,9 @@ func NewResourcesWithTtl(version string, items []types.ResourceWithTtl) Resource
 // from the snapshot may be delivered to the proxy in arbitrary order.
 type Snapshot struct {
 	Resources [types.UnknownType]Resources
+
+	// VersionMap holds the current hash map of all resources in the snapshot. This is only used in delta xDS
```

Comment:
Maybe note that this should remain nil until it's used, at which point it should be instantiated with <name of func>

Reply:
Sure thing!

Reply:
Maybe something like this?
```go
	// VersionMap holds the current hash map of all resources in the snapshot.
	// This field should remain nil until it is used, at which point should be
	// instantiated by setting a snapshot while using the delta xDS protocol.
	// VersionMap is only to be used with delta xDS.
```